transcom/mymove
# B20538 MAIN fix customer search

## Agility ticket
[B20538](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-20538)

## INT PR
[13300](https://github.com/transcom/mymove/pull/13300)

## Summary
Currently, in the loadtest environment, when an officer used the Customer Search, if certain names were searched for the search would crash with an "Oops" response being displayed.  The search was crashing because it was selecting moves where the customer's personal email was empty.  The goal of the change was to eliminate the "Oops" screen and still return the move in the search results.

### How to test
* Customer: 
  - Create a customer with the first name = "Test" and last name = "Test".
  - No move needs to be created.
* Get the "service member id" for the customer
  - SQL:
SELECT u.okta_email
       , sm.id as service_member_id
       , sm.first_name
       , sm.last_name
       , sm.personal_email
  FROM service_members AS sm
       JOIN users u 
         ON u.id = sm.user_id
 where u.okta_email = '20240722215710-66b59eff976a@example.com'
  - Change the okta_email in the where clause to the email login on the login screen
  - The first and last name retrieved by query should match what was entered during customer setup.
* Update service member personal email to null
  - SQL:
update service_members 
   set personal_email = null
 where id = '7fcdfc20-6479-49e2-bc4e-4b5eafeb7a04'
  - set id to the "server_member_id" value retrieved by the previous SQL.
  - run the SQL
* SC: 
  - Login
  - Go the the "Customer Search" tab
  - Select "Name" in the radio button.
  - Search for "Test".
  - The search should finish successfully and the customer you created should be in the list.

## Screenshots
None
